### ISSUE TITLE
Tpetra: Require Teuchos_ENABLE_LONG_LONG_INT:BOOL=ON, at least when GO=long long is enabled

### ISSUE THREAD

**Issue by mhoemmen** [trilinos/Trilinos#2496]
See #2495 for an example of an issue that could be prevented by this.

@trilinos/tpetra 

## Expectations

Tpetra should stop with an error at configure time if `Teuchos_ENABLE_LONG_LONG_INT:BOOL=OFF` but `Tpetra_INST_INT_LONG_LONG:BOOL=ON`.

## Current Behavior

Tpetra currently emits a warning at configure time if `Teuchos_ENABLE_LONG_LONG_INT:BOOL=OFF`.

## Motivation and Context

We want developers and users to get immediate feedback if they set contradictory build options.

## Related Issues
* Related to #2495

**Comment by mhoemmen** [comment]
Hm, it looks like Tpetra already does this:

https://github.com/trilinos/Trilinos/blob/8b80ad7a2c82cb93de2b2892f856717ff3df6aef/packages/tpetra/CMakeLists.txt#L661

```
IF (DEFINED Teuchos_ENABLE_LONG_LONG_INT)
  IF (Tpetra_INST_INT_LONG_LONG AND NOT Teuchos_ENABLE_LONG_LONG_INT)
    MESSAGE (FATAL_ERROR "Tpetra: GlobalOrdinal = long long is enabled in Tpetra but is not enabled in Teuchos. Set Teuchos_ENABLE_LONG_LONG_INT=ON or set Tpetra_INST_INT_LONG_LONG=OFF.  Note that Tpetra requires C++11, and Teuchos_ENABLE_LONG_LONG_INT is ON by default if the compiler supports C++11.  Thus, if you are seeing this message, you probably set Teuchos_ENABLE_LONG_LONG_INT=OFF explicitly.  Please don't do that.  If you don't want to enable GlobalOrdinal = long long in Tpetra, set Tpetra_INST_INT_LONG_LONG=OFF and leave Teuchos alone.")
  ENDIF ()
ENDIF ()
```
Why did this code not catch @brian-kelley 's configure options?

**Comment by brian-kelley** [comment]
@mhoemmen I only had Teuchos_ENABLE_LONG_LONG_INT=OFF, but I did not have Tpetra_INST_INT_LONG_LONG set explicitly. In this case HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT is set to OFF on line 589, and then that becomes the default option for Tpetra_INST_INT_LONG_LONG on line 644. This is why the error condition on line 661 doesn't happen.

**Comment by brian-kelley** [comment]
I guess the only thing to be done is replace WARNING on 587 with FATAL_ERROR (and update the message accordingly).

**Comment by mhoemmen** [comment]
@brian-kelley This suggests that maybe Tpetra hasn't been tested with `Teuchos_ENABLE_LONG_LONG_INT:BOOL=OFF`.  I'm tempted just to make that `WARNING` a `FATAL_ERROR` but then I'll probably get a million angrygrams so perhaps I'll just fix it.

**Comment by brian-kelley** [comment]
@mhoemmen I actually fixed the OpenMP build by removing the line "Trilinos_ENABLE_Serial=ON" that I missed before. Disabling LONG_LONG didn't cause any issues, at least not within Tpetra.

**Comment by mhoemmen** [comment]
@brian-kelley Something is wrong, though; if `HAVE_TPETRA_INST_INT_LONG_LONG_DEFAULT` is `OFF`, that means that `GO = long long` is disabled.

**Comment by brian-kelley** [comment]
Ah, good point.

**Comment by mhoemmen** [comment]
I'm closing this issue and replacing it with #2498; thanks @brian-kelley ! :D

**Comment by mhoemmen** [comment]
PR #2499, just merged into develop, fixes #2498.
